cuda/btx_cudainterval_callbacks.cpp

```diff
@@ -0,0 +1,589 @@
+#include <cassert>
+#include <cstdlib>
+#include <iostream>
+#include <regex>
+#include <tuple>
+#include <string>
+#include <unordered_map>
+
+#include "btx_cudainterval_callbacks.hpp"
+
+#include "context_manager.hpp"
+
+#include <metababel/metababel.h>
+
+
+struct data_s {
+  // store entry state per thread in order to use in exit callback, conditioned
+  // on the cuResult being a non-error
+  std::unordered_map<hpt_t, int64_t> entry_ts;
+  std::unordered_map<hpt_t, std::string> entry_kernel_name;
+  std::unordered_map<hpt_t, size_t> entry_traffic_size;
+
+  // kernels (CUfunction objects) are "created" before use, at which time we save
```

Comment:
Maybe something like "Used to link CUfunction pointer to their ascii name". Maybe the rest is obvious? CuLaunch take a function pointer not an ascii name, so we need to store this info somewhere.